pkgs/applications/networking/browsers/chromium/update.py

```diff
@@ -105,10 +118,35 @@ def get_latest_ungoogled_chromium_build():
     return {
         'channel': 'ungoogled-chromium',
         'version': version,
-        'ungoogled_tag': tag
+        'ungoogled_tag': tag,
+        'os': 'linux'
     }
 
 
+def chrome_mac_url(channel_name):
+    """Maps a channel name to the corresponding universal macOS download URL for Chrome."""
+    if channel_name == 'stable':
+        return 'https://dl.google.com/chrome/mac/universal/stable/CHFA/googlechrome.dmg'
+    return f'https://dl.google.com/chrome/mac/universal/{channel_name}/googlechrome{channel_name}.dmg'
+
+
+def get_latest_chrome_mac_hash(channel_name, version):
+    """Returns the SHA256 hash of the latest macOS build for Chrome."""
+    url = chrome_mac_url(channel_name)
+    hash, path = nix_prefetch_url(url, path=True)
+    # The macOS download doesn't have a known version specific URL
+    # so download the latest and check that the version is what is
+    # expected.
```

Comment:
Oops, that is far from ideal... :o
How well did you check that? And did you look at what Homebrew, etc. are doing (if they have a Chrome package)?

Reply:
Ok, thanks for checking :) Unfortunately this could be a blocker as Nixpkgs channels can be quite delayed and this could cause more problems than it's worth. I haven't checked that part of the implementation yet but I assume that it uses the usual URL + checksum approach and that the usual problems apply. The main problem is that the content of the URL wont always match the checksum (Nixpkgs/NixOS channel too old, etc.) and the build will fail as a result (even caching by Hydra doesn't fully solve this issue and proprietary packages like Google Chrome shouldn't be cached anyway - I'm not sure though if that's implemented correctly).

Anyway, I'm not sure how to move forward with this. It would be best if you could find additional reviewers as this is beyond my knowledge of Nixpkgs policies, etc. Personally, I'm unfortunately against this (unless I missed something and we can avoid those impurity issues) but I'm fine with it if another committer approves it.